libs/vkd3d/state.c

```diff
@@ -1560,6 +1560,69 @@ HRESULT d3d12_root_signature_create_local_static_samplers_layout(struct d3d12_ro
     return S_OK;
 }
 
+HRESULT d3d12_root_signature_create_work_graph_layout(struct d3d12_root_signature *root_signature,
+        VkDescriptorSetLayout *vk_push_set_layout, VkPipelineLayout *vk_pipeline_layout)
+{
+    VkDescriptorSetLayout set_layouts[VKD3D_MAX_DESCRIPTOR_SETS];
+    struct d3d12_bind_point_layout bind_point_layout;
+    VkDescriptorSetLayoutBinding binding;
+    VkPushConstantRange range;
+    bool uses_push_ubo;
+    HRESULT hr;
+
+    /* If we're already using push UBO block, we just need to modify the push range. */
+    /* TODO: Local sampler set. */
+    uses_push_ubo = !!(root_signature->compute.flags & VKD3D_ROOT_SIGNATURE_USE_PUSH_CONSTANT_UNIFORM_BLOCK);
+    range.stageFlags = VK_SHADER_STAGE_COMPUTE_BIT;
+    range.offset = 0;
+    range.size = sizeof(struct vkd3d_shader_node_input_push_signature);
+
+    if (root_signature->root_descriptor_push_mask)
+    {
+        FIXME("The root signature is already using push descriptors, cannot add another push descriptor set. Make sure to use VKD3D_CONFIG=force_raw_va_cbv on NVIDIA.\n");
```

Comment:
Do any of the existing demos hit this path, and is it *likely* that any real shipping apps will?

If so then this sounds like it's going to turn into a bit of a maintenance nightmare with app workarounds...

Reply:
One of them does, indeed. I think the real solution will be to augment the push descriptor set to support both root CBVs + the root params CBV, but that's a significant refactor that is somewhat orthogonal.